### DISCUSSION TITLE
Support for Terraform post-boot service configuration

### DISCUSSION THREAD

**Discussion by sk2** [srl-labs/containerlab#1439] [Ideas]
Hi, we are using containerlab to run service topologies, such as OpenBMP, Netflow, etc.

We'd like to be able to use Terraform to configure these service container after they have booted.
Is it possible to have a method through which an appropriate Terraform compatible description of the container coordinates is provided?

Something conceptually similar to the Ansible plugin would be ideal.

Thanks
Simon

**Comment by hellt** [comment]
Hi @sk2 
when you say "an appropriate Terraform compatible description of the container coordinates is provided" what do you mean exactly? Do you have an example?

**Comment by Therros2023** [comment]
Hi @hellt 

Apologies for the delay on this response, I work with @sk2 and we're wanting to see if you know of anything upcoming with terraform that will have a provider that will support containerlabs.

I know there is a docker provider for terraform, but I'm pretty sure that only supports spinning up containers. Unless I'm wrong it cannot do links between containers/container variables etc. Which is what we were looking at getting terraform to do.

Kind Regards.

**Comment by hellt** [comment]
@Therros2023 
yes, I haven't heard of any TF provider that does links between containers.

**Comment by Therros2023** [comment]
Noted, do you know if this will be coming in the future by chance at some point.

Kind Regards.

**Reply by hellt** [reply]
no plans. The declarative interface is provided by containerlab itself, I do not see why would I want to use TF at this point
